specification/addons/resource-manager/Microsoft.Addons/preview/2018-03-01/addons-swagger.json

```diff
@@ -0,0 +1,438 @@
+        "summary": "Creates or updates the Canonical support plan of type {type} for the subscription.",
+        "description": "",
+        "x-ms-long-running-operation": true,
+        "x-ms-examples": {
+          "SupportPlanTypes_CreateOrUpdate": {
+            "$ref": "./examples/SupportPlanTypes_CreateOrUpdate.json"
+          }
+        },
+        "parameters": [
+          {
+            "$ref": "#/parameters/ApiVersionParameter"
+          },
+          {
+            "$ref": "#/parameters/SubscriptionIdParameter"
+          },
+          {
+            "$ref": "#/parameters/SupportProviderName"
+          },
+          {
+            "$ref": "#/parameters/CanonicalSupportPlanType"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "OK",
+            "schema": {
+              "$ref": "#/definitions/CanonicalSupportPlanResponseEnvelope"
+            }
+          },
+          "201": {
+            "description": "Created",
+            "schema": {
+              "$ref": "#/definitions/CanonicalSupportPlanResponseEnvelope"
+            }
+          },
+          "404": {
+            "description": "Not Found"
+          },
+          "default": {
+            "description": "'404' - 'Not Found'.\n'400' - 'Bad Request'",
+            "schema": {
+              "$ref": "#/definitions/ErrorDefinition"
+            }
+          }
+        }
+      },
+      "delete": {
+        "operationId": "SupportPlanTypes_Delete",
+        "summary": "Cancels the Canonical support plan of type {type} for the subscription.",
+        "description": "",
+        "x-ms-long-running-operation": true,
+        "x-ms-examples": {
+          "SupportPlanTypes_Delete": {
+            "$ref": "./examples/SupportPlanTypes_Delete.json"
+          }
+        },
+        "parameters": [
+          {
+            "$ref": "#/parameters/ApiVersionParameter"
+          },
+          {
+            "$ref": "#/parameters/SubscriptionIdParameter"
+          },
+          {
+            "$ref": "#/parameters/SupportProviderName"
+          },
+          {
+            "$ref": "#/parameters/CanonicalSupportPlanType"
+          }
+        ],
+        "responses": {
+          "202": {
+            "description": "Accepted",
+            "headers": {
+            },
+            "schema": {
+              "$ref": "#/definitions/CanonicalSupportPlanResponseEnvelope"
+            }
+          },
+          "204": {
+            "description": "No Content"
+          },
+          "default": {
+            "description": "BadRequest",
+            "schema": {
+              "$ref": "#/definitions/ErrorDefinition"
+            }
+          }
+        }
+      }
+    },
+    "/subscriptions/{subscriptionId}/providers/Microsoft.Addons/supportProviders/canonical/listSupportPlanInfo": {
+      "post": {
+        "tags": [
+          "AddonStatus"
+        ],
+        "operationId": "AddonInformationForSubscriptionOperation",
```

Comment:
operation ID? I'm confused now: Does this (2018) API version also have `CanonicalSupportPlan`? I see an example called "CanonicalSupportPlanTypes_Get.json" under the 2018 folder, but it's not used anywhere (and uses the "will..." booleans).

Reply:
@olydis That example shouldn't be there. Will remove it. Thanks. Apologies for the confusion.

So in the review board meeting, we were told to change the [GET](https://github.com/rajshah11/azure-rest-api-specs/blob/swagger-for-addons/specification/addons/resource-manager/Microsoft.Addons/preview/2017-05-15/Addons.json#L202) call to a [POST](https://github.com/rajshah11/azure-rest-api-specs/blob/swagger-for-addons/specification/addons/resource-manager/Microsoft.Addons/preview/2018-03-01/addons-swagger.json#L204). They also suggested to rename the endpoint to listSupportPlanInfo and replace the "will..." booleans with a three state flag `oneTimeCharge` with `[ "no", "onEnabled", "onReenabled" ]` as the states. These changes are reflected in the new version. Thanks!

Reply:
were you also asked to name the operation `AddonInformationForSubscriptionOperation`? That looks odd to me - who did you do the API review with, I'll gladly check with them directly 🙂

Reply:
nothing specific on the operation name was discussed. @olydis - please recommend a reasonable name.I can think of something like supportPlanTypes_listInfo